merkle-tree/indexed/src/array.rs

```diff
@@ -167,14 +167,17 @@ where
         self.append(&init_value)
     }
 
-    /// Returns the index of the low element for the given `value`, which
-    /// **should not** be the part of the array.
+    /// Returns the index of the low element for the given `value`, which is
+    /// not yet the part of the array.
     ///
     /// Low element is the greatest element which still has lower value than
     /// the provided one.
     ///
     /// Low elements are used in non-membership proofs.
-    pub fn find_low_element_index(&self, value: &BigUint) -> Result<I, IndexedMerkleTreeError> {
+    pub fn find_low_element_index_for_nonexistent(
+        &self,
+        value: &BigUint,
+    ) -> Result<I, IndexedMerkleTreeError> {
         // Try to find element whose next element is higher than the provided
         // value.
         for (i, node) in self.elements[..self.len() + 1].iter().enumerate() {
```

Comment:
This method now should throw an error if ``*value == node.value``  because then the element already exists right?

Reply:
Done